python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -127,6 +127,9 @@ def compute_partition_keys(self) -> AbstractSet[str]:
             for akpk in self._compatible_subset.asset_partitions
         }
 
+    def compute_asset_partitions(self) -> AbstractSet[AssetKeyPartitionKey]:
+        return self._compatible_subset.asset_partitions
+
```

Comment:
I think we should consider a scarier name for this to indicate that you should *really* understand the constraints of the callsite before calling it, as this will materialize the entire partition range in memory.

Reply:
Is that functionally all that different from the potential impact of other types of `compute_*()` functions? Many of those involve actual round trips to the database, which is comparatively expensive to iterating through a ton of time window partitions.

Reply:
I basically think operation should be illegal except in very specific contexts, and it should be named accordinly.

Reply:
hm I disagree -- I think this is a fairly standard operation that is a sub routine of plenty of other similarly-named operations (e.g. intersecting two AssetSlices requires computing all the keys in each slice), and so if we were to make this illegal, we'd need to make intersections and unions etc. illegal as well

Reply:
I feel pretty strongly this. We can change this later, but for now I would prefer the name

`expensively_materialize_partition_key_range_in_memory` or something ridiculous like that is very easily greppable. This is not a public API so we can codemod it later.

But that will make people think about this in the short-term and make sure that we are in a context where there aren't huge partition ranges.

@gibsondan keeps on running into this in `py-spy` runs so I consider this "on fire" at the moment.

Reply:
Ok I updated the name to `compute_asset_partitions_EXPENSIVE`, which I think is a sufficient deterrent for now, and we can move this discussion to a future diff